### PR TITLE
#166240843 article's statistics

### PR DESCRIPTION
**Pull request by alainburindi**
##### What does this PR do?
Have the ``` article's statistics``` working
##### Description of Task to be completed?
* Allow user to see the article's statistics
##### Have the following endpoints working  
* GET ```/api/articles/:slug/stats```: search for articles according to URL queries
##### How should this be manually tested?
* Clone the repos
* Install dependencies ```npm install```
* Run ```npm run dev```
* Get specific article's stats passing the slug as a parameter
##### What are the relevant pivotal tracker stories?
[#166240843](https://www.pivotaltracker.com/story/show/166240843)

Screen Shot 2019-07-10 at 10 44 12

### CODE REVIEW

**Review by Ramadhan0 — COMMENTED** [review]
@alainburindi please add screenshots

File: src/api/models/article.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default (sequelize, DataTypes) => {
       title: { type: DataTypes.STRING, allowNull: false },
       description: { type: DataTypes.TEXT, allowNull: false },
       body: { type: DataTypes.TEXT, allowNull: false },
-      slug: { type: DataTypes.STRING, allowNull: false },
+      slug: { type: DataTypes.STRING, allowNull: false, unique: true },
```
Comment (murediane) [inline src/api/models/article.js:19]: Why did you add unique on slug yet the algorithm for it is to make it unique ?

Comment (alainburindi) [reply src/api/models/article.js:19]: we need to make sure that the database reflects the reality

File: src/api/migrations/20190702140355-create-comment.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        key: 'id'
+      },
+      onDelete: 'CASCADE',
+      onUpdate: 'CASCADE',
+      allowNull: false
+    },
+    commentBody: {
+      type: Sequelize.STRING
+    },
+    iteration: {
```
Comment (MemunaHaruna) [inline src/api/migrations/20190702140355-create-comment.js]: @alainburindi what does iteration stand for? Also, does your task include creating article comments?

File: src/api/migrations/20190708141337-create-stat.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+export default {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Reads', {
```
Comment (MemunaHaruna) [inline src/api/migrations/20190708141337-create-stat.js:3]: @alainburindi Instead of multiple models/tables for the stats, what do you think about a stats table that just holds the details of the article and then all stats such as readCount, shareCount, commentCount etc for each article is stored in that one table.

These stats would then get incremented every time an article is viewed, shared or commented on.

File: src/api/migrations/20190702140355-create-comment.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+export default {
```
Comment (premlogic) [inline src/api/migrations/20190702140355-create-comment.js]: @alainburindi the comment table is already there. Try to use it

File: src/api/models/comment.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+export default (sequelize, DataTypes) => {
```
Comment (premlogic) [inline src/api/models/comment.js]: @alainburindi there is a model for comments. try to use it.

**Review by nkpremices — CHANGES_REQUESTED** [review]
Good work @alainburindi 
I left some comments for you

**Comment by alainburindi** [comment]
> https://github.com/andela/savage_rangers-ah/pull/72#discussion_r302096809

@MemunaHaruna, It's a good idea but that means that I will have to modify commentcontroller and sharecontroller which I don't have so far

**Review by nkpremices — APPROVED** [review]

**Review by Ramadhan0 — APPROVED** [review]

**Review by murediane — APPROVED** [review]

File: tests/article.stat.test.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  it('Should send article\'s stat', (done) => {
+    chai
+      .request(server)
+      .get('/api/articles/How-to-create-sequalize-seeds/stats')
+      .end((err, res) => {
+        res.body.article.stats.should.include(expectedStat);
+        done();
+      });
+  });
+  it('Should increase the number of reads', (done) => {
```
Comment (MemunaHaruna) [inline tests/article.stat.test.js:43]: @alainburindi but you're not checking that the number of reads has actually increased
